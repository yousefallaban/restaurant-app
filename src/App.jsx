import { RestaurantSearchPage } from '@/pages/Index.js';

function App() {
  return <RestaurantSearchPage />;
}

export default App;
