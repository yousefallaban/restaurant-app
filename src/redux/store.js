import { configureStore } from '@reduxjs/toolkit';
import createSagaMiddleware from 'redux-saga';
import restaurantReducer from './slices/restaurantSlice';
import { watchFetchRestaurants } from './sagas/restaurantSaga';
import { all } from 'redux-saga/effects';

function* rootSaga() {
  yield all([watchFetchRestaurants()]);
}

const sagaMiddleware = createSagaMiddleware();

const store = configureStore({
  reducer: {
    restaurants: restaurantReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({ thunk: false }).concat(sagaMiddleware),
});

sagaMiddleware.run(rootSaga);

export default store;
