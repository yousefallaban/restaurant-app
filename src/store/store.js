import { configureStore, combineReducers } from '@reduxjs/toolkit';
import createSagaMiddleware from 'redux-saga';

// Todo : Replace with actual reducers
const dummyReducer = (state = {}, action) => state;

const rootReducer = combineReducers({
  dummy: dummyReducer,
});

const sagaMiddleware = createSagaMiddleware();

const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({ thunk: false }).concat(sagaMiddleware),
});

sagaMiddleware.run(function* rootSaga() {
  // Todo: Replace with actual sagas
});

export default store;
