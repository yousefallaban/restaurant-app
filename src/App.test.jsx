import { render, screen } from '@testing-library/react';
import App from './App';
import { Provider } from 'react-redux';
import store from '@/redux/store'; // Adjust as needed

describe('App Component', () => {
  it('should have a heading element', () => {
    render(
      <Provider store={store}>
        <App />
      </Provider>
    );
    expect(screen.getByRole('heading')).toBeInTheDocument();
  });
});
