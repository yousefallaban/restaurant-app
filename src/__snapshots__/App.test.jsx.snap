// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should have a heading element 1`] = `
<DocumentFragment>
  <div
    class="gridLayout"
  >
    <aside
      class="sidebar"
    >
      <div>
        <form
          class="container"
        >
          <div
            class="inputGroup"
          >
            <div
              class="inputWrapper"
            >
              <input
                class="inputField"
                id="postcode-«r0»"
                name="postcode"
                placeholder="Enter UK Postcode"
                type="text"
                value=""
              />
            </div>
          </div>
          <button
            class="buttonBase filled"
            type="submit"
          >
            Search
          </button>
        </form>
      </div>
    </aside>
    <main
      class="mainContent"
    >
      <div
        style="display: flex; gap: 1rem;"
      >
        <div
          class="inputGroup"
        >
          <div
            class="inputWrapper"
          >
            <input
              class="inputField"
              placeholder="Filter by name or cuisine or place"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="dropdownGroup"
        >
          <div
            class="dropdownWrapper"
          >
            <select
              class="dropdownField"
            >
              <option
                value=""
              >
                Sort by
              </option>
              <option
                value="rating"
              >
                Reviews
              </option>
              <option
                value="driveDistanceMeters"
              >
                Delivery time
              </option>
              <option
                value="minimumDeliveryValue"
              >
                Delivery cost
              </option>
            </select>
            <span
              aria-hidden="true"
              class="dropdownIcon"
            >
              ▾
            </span>
          </div>
        </div>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
